sd-finetuner-workflow/sd-finetuner/finetuner.py

```diff
@@ -232,128 +294,6 @@ def get_gpu_ram() -> str:
-        self.image_files = []
-        [
-            self.image_files.extend(glob.glob(f"{data_root}/" + "*." + e))
-            for e in ext
-        ]
-
-        self.examples = {}
-        self.hashes = []
-        for i in self.image_files:
-            hash = i[len(f"{data_root}/") :].split(".")[0]
-            self.examples[hash] = {
-                "image": i,
-                "text": f"{data_root}/{hash}.txt",
-            }
-            self.hashes.append(hash)
-
-        self.size = size
-        self.interpolation = {
-            "bilinear": PIL.Image.Resampling.BILINEAR,
-            "bicubic": PIL.Image.Resampling.BICUBIC,
-            "lanczos": PIL.Image.Resampling.LANCZOS,
-        }[interpolation]
-
-        self.transforms = torchvision.transforms.Compose(
-            [
-                torchvision.transforms.Lambda(
-                    lambda x: resize_image(
-                        x, (self.size, self.size), self.interpolation
-                    )
-                ),
-                torchvision.transforms.CenterCrop(self.size)
-                if args.center_crop
-                else torchvision.transforms.RandomCrop(self.size),
-                torchvision.transforms.RandomHorizontalFlip(p=0.5),
-                torchvision.transforms.ToTensor(),
-                torchvision.transforms.Normalize([0.5], [0.5]),
-            ]
-        )
-
-        self.ucg = ucg
-        self.tokenizer = tokenizer
-
-    def get_caption(self, i: int) -> str:
-        example = self.examples[self.hashes[i]]
-        caption = open(example["text"], "r").read()
-        caption = (
-            caption.replace("  ", " ").replace("\n", " ").lstrip().rstrip()
-        )
-        return caption
-
-    def __len__(self) -> int:
-        return len(self.image_files)
-
-    def __getitem__(self, i: int) -> dict:
-        image = {}  # pixel values, input ids
-        try:
-            image_file = self.examples[self.hashes[i]]["image"]
-            with open(image_file, "rb") as f:
-                image_pil = Image.open(f).convert("RGB")
-                image["pixel_values"] = self.transforms(image_pil)
-            text_file = self.examples[self.hashes[i]]["text"]
-            with open(text_file, "rb") as f:
-                text = f.read().decode("utf-8")
-                text = (
-                    text.replace("  ", " ").replace("\n", " ").lstrip().rstrip()
-                )
-                image["input_ids"] = self.tokenizer(
-                    text,
-                    max_length=self.tokenizer.model_max_length,
-                    padding="do_not_pad",
-                    truncation=True,
-                ).input_ids
-        except Exception as e:
-            print(
-                f'Error with {self.examples[self.hashes[i]]["image"]} -- {e} -- skipping {i}'
-            )
-            return None
-
-        if random.random() < self.ucg:
-            image["caption"] = ""
```

Comment:
I've removed UCG as an argument in my changes because I noticed it wasn't properly implemented. This if statement sets a `caption` value to an empty string but the key that is used everywhere else in the code is `input_ids`.

@harubaru - Is just emptying the `input_ids` enough to drop text conditioning or would we need extra logic around it in the step function when using the UNet model?

Reply:
I cross-referenced the code based off of the original implementation which also just dropped text conditioning only:
https://github.com/pesser/stable-diffusion/blob/57eea7dfc2cdd8cadae77ab1c391f956d46f69bd/ldm/models/diffusion/ddpm.py#L396